docs/eureka/custom-erc20-integration.mdx

```diff
@@ -0,0 +1,111 @@
+
+Many teams bridging through the Cosmos Hub, however, want ownership and control over their ERC20 contracts on Ethereum. Since `IBCERC20` is managed by the `ICS20Transfer` contract and isn't customizable, direct ownership isn't possible.
+
+To address this, we allow teams to deploy custom ERC20 contracts—provided they implement a simple interface that lets the `ICS20Transfer` contract mint and burn tokens.
+
+## Benefits
+
+The benefits of this approach include:
+
+<CardGroup cols={2}>
+  <Card title="Custom Metadata" icon="tag">
+    Customize metadata and token naming on deployment. Tokens will not initially be named `ibc/transfer/channel-0...` and can be represented with the name they are recognized by.
+  </Card>
+  <Card title="Contract Verification" icon="shield-check">
+    Easier contract verification on Etherscan. Each project can easily verify the ERC20 contract deployed for their project to increase trust by associating the token with the official project domain.
+  </Card>
+  <Card title="Improved CEX Listing" icon="dollar-sign">
+    This is likely to result in easier CEX listing and generally increased trust in the bridged asset.
+  </Card>
+  <Card title="Full Control" icon="gear">
+    Complete ownership and control over the ERC20 contract representing your token on Ethereum.
+  </Card>
+</CardGroup>
+
+## Requirements
+
+To replace the default `IBCERC20`, your custom ERC20 contract must implement the [`IMintableAndBurnable`](https://github.com/cosmos/solidity-ibc-eureka/blob/main/contracts/interfaces/IMintableAndBurnable.sol) interface:
+
+```solidity
+interface IMintableAndBurnable {
+    /// @notice Mint new tokens to the Escrow contract
+    /// @dev This function can only be called by an authorized contract (e.g., ICS20)
+    /// @dev This function needs to allow minting tokens to the Escrow contract
+    /// @param mintAddress Address to mint tokens to
+    /// @param amount Amount of tokens to mint
+    function mint(address mintAddress, uint256 amount) external;
+
+    /// @notice Burn tokens from the Escrow contract
+    /// @dev This function can only be called by an authorized contract (e.g., ICS20)
+    /// @dev This function needs to allow burning of tokens from the Escrow contract
+    /// @param mintAddress Address to burn tokens from
+    /// @param amount Amount of tokens to burn
+    function burn(address mintAddress, uint256 amount) external;
+}
+```
+
+For an example implementation of this interface, you can refer to the [`RefImplIBCERC20.sol`](https://github.com/cosmos/solidity-ibc-eureka/blob/main/test/solidity-ibc/utils/RefImplIBCERC20.sol) contract in the `solidity-ibc-eureka` repository.
+
+### Access Control Requirements
+
+These functions must be callable by the proxy of the `ICS20Transfer` contract:
```

Comment:
Might be worth noting that nobody else should be able to call it either :)
If a team leaves these functions open to other to call, their token is compromised.